configs/mir_config.yaml

```diff
@@ -111,3 +111,19 @@ plugins:
     # Examples of additional fields for cart detection module
     io_name: "MiR internal IOs"
     latch_io: 2
+  rmf_wait_until:
+    module: "fleet_adapter_mir_actions.rmf_wait_until"
+    actions: ["wait_until"]
+    timeout: 1800 # Optional, in seconds. Timeout if there is no move-off signal received, after which robot will move off regardless. Default to 60 seconds.
+    update_gap: 30 # Optional, in seconds. The fleet adapter will log an update using this interval. Default to 30 seconds.
+    signal_type: # Defines the default configs for various move off signal types. If this field is not provided, the robot will wait the full duration of the default timeout. Signal configs can be overridden by task description.
+      # Currently three signal types are supported: [mission, plc, custom].
+      mission:
+        mission_name: "some_mission_name" # If this mission is found on the robot, it will be queued when the action starts, and the robot will move off when the mission is completed.
+        retry_count: 10 # Optional field for signal type "mission". The fleet adapter will re-attempt queueing the mission for this number of tries.
+        resubmit_on_abort: False # Optional field for signal type "mission". Set to True to resubmit mission if the mission gets aborted by the robot. Default to False.
+      plc:
+        register: 20 # Fill in with PLC register number. Robot moves off when this PLC register returns True
+      custom:
+        module: "fleet_adapter_mir_actions.rmf_move_off_on_alert" # Fill in with path to custom module written. Robot moves off when module provides the user-defined move off signal.
+      default: "custom" # Specifies a default signal type out of those listed above. If the task description does not provide any signal config, the fleet adapter will use this default signal type.
```

Comment:
This is a really interesting suggestion and can be incorporated very easily. I can see the main pro being having everything configured at launch, such that users who would like 10 different missions to be triggered for various tasks won't have to constantly overwrite default values with their task descriptions. In fact, this helps to reduce the complexity of task descriptions - instead of allowing users to pick and choose the values they want to overwrite, I'd limit their options to either A) pick an existing configured signal type, B) provide a mission name/PLC register (no more submitting "PLC" task without specifying the register value), or C) pick the default signal like before.

From experience I feel like high level of configurability never hurts (as long as they are clear), and more often than not it is requested when we least expect it 🤣  I'll work on adding this.

Reply:
[6284a28](https://github.com/open-rmf/fleet_adapter_mir/pull/39/commits/6284a28aaad8aacdf6c916a3861128606fa315f9)